conanfile.py

```diff
@@ -0,0 +1,247 @@
+    def build_requirements(self):
+        if self.options.enable_testing:
+            self.test_requires("gtest/1.10.0")
+            self.test_requires("cpputest/4.0")
+
+    def configure(self):
+        if self.options.build_all:
+            for opt, val in self.options.values.items():
+                if opt.startswith('build_'):
+                    setattr(self.options, opt, True)
+        if not self.options.celix_cxx:
+            self.options.build_cxx_remote_service_admin = False
+            self.options.build_promises = False
+            self.options.build_pushstreams = False
+        if not self.options.build_cxx_remote_service_admin:
+            self.options.build_cxx_rsa_integration = False
+        if not self.options.enable_testing:
+            self.options.build_pubsub_integration = False
+        if not self.options.build_device_access:
+            self.options.build_device_access_example = False
+        if not self.options.build_log_service:
+            self.options.build_syslog_writer = False
+        if not self.options.build_pubsub:
+            self.options.build_pubsub_psa_zmq = False
+            self.options.build_pubsub_examples = False
+            self.options.build_pubsub_integration = False
+            self.options.build_pubsub_psa_tcp = False
+            self.options.build_pubsub_psa_udp_mc = False
+            self.options.build_pubsub_psa_ws = False
+            self.options.build_pubsub_discovery_etcd = False
+        if not self.options.build_remote_service_admin:
+            self.options.build_rsa_remote_service_admin_dfi = False
+            self.options.build_rsa_discovery_configured = False
+            self.options.build_rsa_discovery_etcd = False
+        if not self.options.build_shell:
+            self.options.build_remote_shell = False
+            self.options.build_shell_bonjour = False
+            self.options.build_shell_tui = False
+            self.options.build_shell_wui = False
+        if not self.options.celix_install_deprecated_api:
+            self.options.build_shell_bonjour = False
+
+    def requirements(self):
+        self.requires("libffi/[>=3.2.1 <4.0.0]")
+        self.options['libffi'].shared = True
+        self.requires("jansson/[>=2.12 <3.0.0]")
+        self.options['jansson'].shared = True
+        self.requires("libcurl/[>=7.64.1 <8.0.0]")
+        self.options['libcurl'].shared = True
+        self.requires("zlib/[>=1.2.8 <2.0.0]")
+        self.options['zlib'].shared = True
+        self.requires("libuuid/1.0.3")
+        self.options['libuuid'].shared = True
+        self.requires("libzip/[>=1.7.3 <2.0.0]")
+        self.options['libzip'].shared = True
+        self.options['openssl'].shared = True
+        self.options['libxml2'].shared = True
+        if self.options.enable_testing:
+            self.options['gtest'].shared = True
+            if self.options.enable_address_sanitizer:
+                self.options["cpputest"].with_leak_detection = False
```

Comment:
Without Leak Detection Plugin, clang address sanitizer now works with cpputest, the following is addressed:
https://github.com/apache/celix/blob/9cf8bc39c1a8916fbdda54eac59b24065f0fe3b2/.github/workflows/ubuntu.yml#L16